UnitWorld_client_gui/clientgui/userControls/ServerCommander.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "shared/transfers/PhysicsCommunicationAssembler.h"
+
+#include "shared/communication/messages/MoveMobileUnitsToPositionMessage.h"
+#include "shared/communication/MessageSerializer.h"
+
+#include "shared/game/physics/Vector2D.h"
+
+#include "communications/CommunicationHandler.h"
+
+#include "commons/Guid.hpp"
+
+#include <vector>
+
+namespace uw
+{
+    class ServerCommander
+    {
+    public:
+        ServerCommander(std::shared_ptr<CommunicationHandler> serverCommunicator, std::shared_ptr<PhysicsCommunicationAssembler> physicsCommunicationAssembler, std::shared_ptr<MessageSerializer> messageSerializer) :
+            _serverCommunicator(serverCommunicator),
+            _physicsCommunicationAssembler(physicsCommunicationAssembler),
+            _messageSerializer(messageSerializer)
+        {}
+
+        void moveUnitsToPosition(const std::vector<xg::Guid>& singuityIds, const Vector2D& destination)
+        {
+            const auto communicatedDestination(_physicsCommunicationAssembler->physicsVector2DToCommunicated(destination));
+
+            const auto message(std::make_shared<MoveMobileUnitsToPositionMessage>(singuityIds, communicatedDestination));
+
+            _serverCommunicator->send(_messageSerializer->serialize(std::vector<MessageWrapper> { MessageWrapper(message) }));
```

Comment:
Because the message serializer only way to serialize is to take multiple messages in input, same with its output.

WHere would you cut the line?

Reply:
I would put `_messageSerializer->serialize(std::vector<MessageWrapper> { MessageWrapper(message) })` on a separate line if you mind to maintain the repository for long enough; leave it alone if you plan to throw the repository soon :)

Reply:
> Because the message serializer only way to serialize is to take multiple messages in input, same with its output.

I see, no problem for this then

Reply:
See, there are many lines that might have to be split in 2 or more to gain more readability, but I guess this doesn't matter for this repository

Reply:
I might touch them again, but this will be when I still have a fairly good Idea of the whole code. UnitWorld might not be still alive in two years, but I might edit this line a bunch of times until then.